common/utils.c

```diff
@@ -428,13 +428,24 @@ TDNFYesOrNo(
 
     if(!pArgs->nAssumeYes && !pArgs->nAssumeNo)
     {
-        pr_crit("%s", pszQuestion);
-        while ((opt = getchar()) == '\n' || opt == '\r');
-        opt = tolower(opt);
-        if (opt != 'y' && opt != 'n')
-        {
-            dwError = ERROR_TDNF_INVALID_INPUT;
-            BAIL_ON_TDNF_ERROR(dwError);
+        while(1) {
+            pr_crit("%s", pszQuestion);
+            char buf[256];
+            char *ret;
+
+            ret = fgets(buf, sizeof(buf)-1, stdin);
+            if (ret != buf || strlen(buf) == 0) {
```

Comment:
check buf for non null before calling strlen or you can do:
```
(... || buf && !buf[0])
```

Reply:
Not needed. When `buf == NULL` then `ret != buf`.

Reply:
NP then but still I would suggest avoid strlen function call and checking for non empty char in buf[0], faster and simple way.